packages/ui-components/src/ClaimButton.ts

```diff
@@ -0,0 +1,688 @@
+    const secs = seconds % 60
+    let parts = []
+    if (days > 0) parts.push(String(days).padStart(2, "0"))
+    parts.push(String(hours).padStart(2, "0"))
+    parts.push(String(mins).padStart(2, "0"))
+    parts.push(String(secs).padStart(2, "0"))
+    return parts.join(":")
+  }
+
+  updated(changedProperties: any) {
+    if (changedProperties.has("claimState") && this.claimState === "success") {
+      this.startConfetti()
+    }
+  }
+
+  startConfetti() {
+    this.confettiCanvas = this.shadowRoot!.querySelector(
+      "#confetti-canvas",
+    ) as HTMLCanvasElement
+    if (!this.confettiCanvas) return
+    this.confettiCtx = this.confettiCanvas.getContext("2d")
+    if (!this.confettiCtx) return
+
+    this.confettiCanvas.width = this.confettiCanvas.offsetWidth
+    this.confettiCanvas.height = this.confettiCanvas.offsetHeight
+
+    this.particles = []
+    for (let i = 0; i < 20; i++) {
+      this.particles.push({
+        x: Math.random() * this.confettiCanvas.width,
+        y: -Math.random() * this.confettiCanvas.height,
+        vx: (Math.random() - 0.5) * 0.3,
+        vy: Math.random() * 0.7 + 0.2,
+        color: `hsla(${Math.random() * 360}, 60%, 85%, 0.5)`,
+        size: Math.random() * 10 + 5,
+        phase: Math.random() * Math.PI * 2,
+        rotation: 0,
+        rotationSpeed: (Math.random() - 0.5) * 0.05,
+      })
+    }
+
+    let startTime = performance.now()
+    const duration = 5000
+
+    const animate = (currentTime: number) => {
+      if (!this.confettiCanvas) return
+      if (
+        !this.confettiCtx ||
+        this.claimState !== "success" ||
+        !this.confettiCanvas ||
+        currentTime - startTime > duration
+      ) {
+        this.confettiCtx?.clearRect(
+          0,
+          0,
+          this.confettiCanvas.width,
+          this.confettiCanvas.height,
+        )
+        return
+      }
+
+      this.confettiCtx.clearRect(
+        0,
+        0,
+        this.confettiCanvas.width,
+        this.confettiCanvas.height,
+      )
+
+      this.particles.forEach((p) => {
+        if (!this.confettiCanvas || !this.confettiCtx) return
+        p.y += p.vy
+        p.rotation += p.rotationSpeed
+        p.vy += 0.0008
+        p.phase += 0.05
+        p.vx = (Math.random() - 0.5) * 0.3 + Math.sin(p.phase) * 0.1
+        p.x += p.vx
+        if (p.y > this.confettiCanvas.height) {
+          p.y = -p.size
+          p.x = Math.random() * this.confettiCanvas.width
+        }
+
+        this.confettiCtx.save()
+        this.confettiCtx.translate(p.x + p.size / 2, p.y + p.size / 2)
+        this.confettiCtx.rotate(p.rotation)
+        this.confettiCtx.fillStyle = p.color
+        this.confettiCtx.beginPath()
+        this.confettiCtx.rect(-p.size / 2, -p.size / 2, p.size, p.size)
+        this.confettiCtx.fill()
+        this.confettiCtx.restore()
+      })
+
+      requestAnimationFrame(animate)
+    }
+
+    requestAnimationFrame(animate)
+  }
+
+  render() {
+    return html`
```

Comment:
theres no wagmi/reown context, so you need to add in the docs that it is a requirement no?

Reply:
We don't use wagmi or the re-own AppkitProvider context. it works regardless of the framework used.

The claimbutton follows the ethers (v6) pattern: https://docs.reown.com/appkit/javascript/core/installation#ethers
to trigger the modal we follow the 'actions' pattern: https://docs.reown.com/appkit/javascript/core/installation#ethers

so no dependencies or additional setup is needed.
everything is shipped as part of the package or the iife file.